content/docs/05.developer-guide/03.scripts.md

```diff
@@ -0,0 +1,800 @@
+        beforeCommands:
+          - pip install -r requirements.txt > /dev/null
+        commands:
+          - ls -lt $(python -c "import site; print(site.getsitepackages()[0])") | head -n 20
+```
+
+
+## Using Kestra's prebuilt images
+
+Many data engineering use cases require performing fairly standardized tasks such as: 
+
+- processing data with `pandas` 
+- transforming data with `dbt-core` (*using a dbt adapter for your data warehouse*)
+- making API calls with the `requests` library, etc. 
+
+To solve those common challenges, the [kestra-io/examples](https://github.com/orgs/kestra-io/packages?repo_name=examples) repository provides several **public** Docker images with the latest versions of those common packages. Many  [Blueprints](../04.user-interface-guide/blueprints.md)  use those public images by default. The images are hosted in GitHub Container Registry and follow the naming `ghcr.io/kestra-io/packageName:latest`. 
+
+### Example: running R script in Docker 
+
+Here is a simple example using the `ghcr.io/kestra-io/rdata:latest` Docker image provided by Kestra to analyze the built-in `mtcars` dataset using `dplyr` and `arrow` R libraries:
+
+```yaml
+id: rCars
+namespace: dev
+
+tasks:
+  - id: wdir
+    type: io.kestra.core.tasks.flows.WorkingDirectory
+    tasks:
+      - id: r
+        type: io.kestra.plugin.scripts.r.Script
+        warningOnStdErr: false
+        runner: DOCKER
+        docker:
+          image: ghcr.io/kestra-io/rdata:latest
+        script: |
+          library(dplyr)
+          library(arrow)
+
+          data(mtcars) # Load mtcars data
+          print(head(mtcars))
+
+          final <- mtcars %>%
+            summarise(
+              avg_mpg = mean(mpg),
+              avg_disp = mean(disp),
+              avg_hp = mean(hp),
+              avg_drat = mean(drat),
+              avg_wt = mean(wt),
+              avg_qsec = mean(qsec),
+              avg_vs = mean(vs),
+              avg_am = mean(am),
+              avg_gear = mean(gear),
+              avg_carb = mean(carb)
+            ) 
+          final %>% print()
+          write.csv(final, "final.csv")
+
+          mtcars_clean <- na.omit(mtcars) # remove rows with NA values
+          write_parquet(mtcars_clean, "mtcars_clean.parquet")
+
+      - id: outputs
+        type: io.kestra.core.tasks.storages.LocalFiles
+        outputs:
+          - "*.csv"
+          - "*.parquet"
+```
+
+R libraries are time-consuming to install. From a technical standpoint, you could install custom R packages at runtime as follows:
+
+```yaml
+id: rCars
+namespace: dev
+
+tasks:
+  - id: wdir
+    type: io.kestra.core.tasks.flows.WorkingDirectory
+    tasks:
+      - id: r
+        type: io.kestra.plugin.scripts.r.Script
+        warningOnStdErr: false
+        runner: DOCKER
+        docker:
+          image: ghcr.io/kestra-io/rdata:latest
+        beforeCommands: 
+	        - Rscript -e "install.packages(c('dplyr', 'arrow'))" > /dev/null 2>&1
```

Comment:
Again a strangely rendered stuff

Reply:
agree, this is the default indentation when adding beforeCommands in the UI -- at least used to - fixing manually, well spotted